api/grpcserver.go

```diff
@@ -431,13 +449,45 @@ func (svr *GRPCServer) GetLogs(ctx context.Context, in *iotexapi.GetLogsRequest)
 }
 
 // StreamBlocks streams blocks
-func (svr *GRPCServer) StreamBlocks(in *iotexapi.StreamBlocksRequest, stream iotexapi.APIService_StreamBlocksServer) error {
-	return svr.coreService.StreamBlocks(stream)
+func (svr *GRPCServer) StreamBlocks(_ *iotexapi.StreamBlocksRequest, stream iotexapi.APIService_StreamBlocksServer) error {
+	errChan := make(chan error)
+	defer close(errChan)
+	if _, err := svr.coreService.AddListenerResponder(NewGRPCBlockListener(
+		func(resp interface{}) error {
+			return stream.Send(resp.(*iotexapi.StreamBlocksResponse))
+		},
+		errChan,
+	)); err != nil {
+		return status.Error(codes.Internal, err.Error())
+	}
+	err := <-errChan
```

Comment:
it needs to be a `for { select }` loop like in original code?
otherwise the API server will block here

Reply:
as far as I recall, only the for loop way will work